src/files/files-list/FilesList.js

```diff
@@ -41,16 +37,29 @@ export const FilesList = ({
   const [allFiles, setAllFiles] = useState(mergeRemotePinsIntoFiles(files, remotePins))
   const listRef = useRef()
   const filesRefs = useRef([])
+  const [existingFiles, setExistingFiles] = useState(null)
 
   const [{ canDrop, isOver, isDragging }, drop] = useDrop({
     accept: NativeTypes.FILE,
-    drop: (_, monitor) => {
+    drop: async (_, monitor) => {
       if (monitor.didDrop()) {
         return
       }
       const { filesPromise } = monitor.getItem()
 
-      addFiles(filesPromise, onAddFiles)
+      const filesToAdd = normalizeFiles((await filesPromise) || [])
+
+      const filePaths = files.map(f => f.path.replace('/files/', ''))
+      const existingFiles = filesToAdd.filter(({ path }) => filePaths.includes(path))
+
+      if (existingFiles.length) {
+        setExistingFiles(existingFiles)
```

Comment:
What if the same file is dropped twice, it would produce same CID but second drop would still add cid suffix (if user approves) even though there is no real conflict.

Reply:
It overrides the duplicated file (the one that has the appended CID), I believe it's fine since it required user interaction and awareness for that to happen?